### ISSUE TITLE
Unable to connect to inverter

### ISSUE THREAD

**Issue by JRascagneres** [cdpuk/givenergy-local#69]
Likely a new install issue but would like to open an issue so it can be tracked.

## Version of the custom_component
v1.9.0

## Configuration

![image](https://github.com/cdpuk/givenergy-local/assets/3264971/917123fd-0573-4169-9e5e-b42a55caabe1)
```
Failed to validate inverter configuration
Traceback (most recent call last):
  File "/config/custom_components/givenergy_local/config_flow.py", line 48, in async_step_user
    serial_no = await read_inverter_serial(self.hass, user_input)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/givenergy_local/config_flow.py", line 27, in read_inverter_serial
    serial_no: str = plant.inverter.inverter_serial_number
                     ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/givenergy_modbus/model/plant.py", line 32, in inverter
    return Inverter.from_orm(self.inverter_rc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 577, in from_orm
    values, fields_set, validation_error = validate_model(cls, obj)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 1055, in validate_model
    value = input_data.get(field.alias, _missing)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/givenergy_modbus/model/register_getter.py", line 15, in get
    self.get('inverter_serial_number_1_2'),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/givenergy_modbus/model/register_getter.py", line 69, in get
    return getattr(self._obj, key, default)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/givenergy_modbus/model/register_cache.py", line 26, in __getattr__
    val = self[register]
          ~~~~^^^^^^^^^^
KeyError: HR:013
```

```
Did not receive expected response type: ReadInputRegistersResponse != ReadHoldingRegistersResponse
Did not receive expected response type: ReadHoldingRegistersResponse != ReadInputRegistersResponse
Did not receive expected response type: ReadHoldingRegistersResponse != ModbusIOException
Did not receive expected response type: ReadInputRegistersResponse != ModbusIOException
```

```
Modbus Error: [Input/Output] Modbus Error: [Invalid Message] No response received, expected at least 164 bytes (0 received)
Modbus Error: [Input/Output] No Response received from the remote unit/Unable to decode response
NoneType: None
```

## Describe the bug
Doesn't connect and allow integration to start

I attempted to run the script and ended up with:
```
python .\debug.py 192.168.4.70 
root INFO Performing full refresh...
givenergy_modbus INFO Sending request 4/ReadInputRegistersRequest({check: 0xd1d5, slave_address: 0x0032, base_register: 0x0000, register_count: 0x003c})
givenergy_modbus INFO Decoded response 4/ReadInputRegistersResponse({check: 0x85d7, inverter_serial_number: FA2323G963, base_register: 0x00b4, register_count: 0x003c, data_adapter_serial_number: WH2323G963, padding: 0x008a, slave_ad
dress: 0x0031})
givenergy_modbus ERROR Returned base register (180) does not match that from request (0).
givenergy_modbus INFO Sending request 4/ReadInputRegistersRequest({check: 0xf795, slave_address: 0x0032, base_register: 0x00b4, register_count: 0x003c})
givenergy_modbus INFO Decoded response 3/ReadHoldingRegistersResponse({check: 0x9701, inverter_serial_number: FA2323G963, base_register: 0x0000, register_count: 0x003c, data_adapter_serial_number: WH2323G963, padding: 0x008a, slave_
address: 0x0031})
givenergy_modbus ERROR Did not receive expected response type: ReadInputRegistersResponse != ReadHoldingRegistersResponse
givenergy_modbus INFO Sending request 3/ReadHoldingRegistersRequest({check: 0x1160, slave_address: 0x0032, base_register: 0x0000, register_count: 0x003c})
givenergy_modbus INFO Decoded response 3/ReadHoldingRegistersResponse({check: 0xfc99, inverter_serial_number: FA2323G963, base_register: 0x003c, register_count: 0x003c, data_adapter_serial_number: WH2323G963, padding: 0x008a, slave_
address: 0x0031})
givenergy_modbus ERROR Returned base register (60) does not match that from request (0).
givenergy_modbus INFO Sending request 3/ReadHoldingRegistersRequest({check: 0x1da0, slave_address: 0x0032, base_register: 0x003c, register_count: 0x003c})
givenergy_modbus INFO Decoded response 3/ReadHoldingRegistersResponse({check: 0x093d, inverter_serial_number: FA2323G963, base_register: 0x0078, register_count: 0x003c, data_adapter_serial_number: WH2323G963, padding: 0x008a, slave_
address: 0x0031})
givenergy_modbus ERROR Returned base register (120) does not match that from request (60).
givenergy_modbus INFO Sending request 3/ReadHoldingRegistersRequest({check: 0x08e0, slave_address: 0x0032, base_register: 0x0078, register_count: 0x003c})
givenergy_modbus INFO Decoded response 3/ReadHoldingRegistersResponse({check: 0x07ad, inverter_serial_number: FA2323G963, base_register: 0x00f0, register_count: 0x003c, data_adapter_serial_number: WH2323G963, padding: 0x008a, slave_
address: 0x0031})
givenergy_modbus ERROR Returned base register (240) does not match that from request (120).
root INFO Dumping inverter registers...
{}
root INFO Decoding and printing inverter data...
Traceback (most recent call last):
  File "D:\Documents\Programming\givenergy-local\scripts\debug.py", line 68, in <module>
    main()
  File "D:\Documents\Programming\givenergy-local\scripts\debug.py", line 64, in main
    debugger.display_decoded_data()
  File "D:\Documents\Programming\givenergy-local\scripts\debug.py", line 45, in display_decoded_data
    print(self.plant.inverter.json(indent=2))
  File "D:\Documents\Programming\givenergy-local\venv\lib\site-packages\givenergy_modbus\model\plant.py", line 32, in inverter
    return Inverter.from_orm(self.inverter_rc)
  File "pydantic\main.py", line 577, in pydantic.main.BaseModel.from_orm
  File "pydantic\main.py", line 1055, in pydantic.main.validate_model
  File "D:\Documents\Programming\givenergy-local\venv\lib\site-packages\givenergy_modbus\model\register_getter.py", line 15, in get
    self.get('inverter_serial_number_1_2'),
  File "D:\Documents\Programming\givenergy-local\venv\lib\site-packages\givenergy_modbus\model\register_getter.py", line 69, in get
    return getattr(self._obj, key, default)
  File "D:\Documents\Programming\givenergy-local\venv\lib\site-packages\givenergy_modbus\model\register_cache.py", line 26, in __getattr__
    val = self[register]
KeyError: HR:013
```

**Comment by cdpuk** [comment]
Please could you confirm your inverter model & generation, battery model, and firmware versions for both?

This is probably an incompatibility with newer firmware, which is being worked on under #61.

**Comment by cdpuk** [comment]
Closing due to lack of detail, though likely to be resolved by #74
